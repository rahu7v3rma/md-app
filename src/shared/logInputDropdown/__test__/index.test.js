import React from 'react';
import renderer, { act } from 'react-test-renderer';

import LogInputDropdown from '@/shared/logInputDropdown';

const mockOpen = jest.fn();
const mockClose = jest.fn();
const mockOnSelect = jest.fn();

let tree;
beforeEach(async () => {
    tree = await act(() =>
        renderer.create(
            <LogInputDropdown
                fieldName="DropDown Test"
                selectedValue={1}
                labelKey="name"
                valueKey="id"
                onSelect={mockOnSelect}
                options={[
                    {
                        id: 1,
                        name: 'choice1'
                    },
                    {
                        id: 2,
                        name: 'choice2'
                    }
                ]}
                onOpen={mockOpen}
                onClose={mockClose}
            />
        )
    );
});

describe('test log input dropdown', () => {
    it('snapshot test', async () => {
        expect(tree).toMatchSnapshot();
    });

    it('renders without crashing', async () => {
        expect(tree.root).toBeTruthy();
    });

    it('dropdown items are initially hidden', async () => {
        expect(tree.root.findAllByProps({ testID: 'choice1' }).length).toEqual(
            0
        );
    });

    it('clicking LogInput opens the dropdown', async () => {
        await act(() => {
            tree.root.findByProps({ value: 'choice1' }).props.onPress();
        });
        expect(
            tree.root.findAllByProps({ testID: 'choice2' }).length
        ).toBeGreaterThan(0);
    });

    it('Clicking a dropdown item triggers onSelect callback', async () => {
        await act(() => {
            tree.root.findByProps({ value: 'choice1' }).props.onPress();
        });
        await act(() => {
            tree.root.findByProps({ testID: 'choice2' }).props.onPress();
        });
        expect(mockOnSelect).toBeCalled();
    });

    it('clicking outside the dropdown closes it', async () => {
        await act(() => {
            tree.root.findByProps({ value: 'choice1' }).props.onPress();
        });
        await act(() =>
            tree.root
                .findByProps({ hardwareAccelerated: false })
                .props.onRequestClose()
        );
        await act(() => {
            tree.root.findByProps({ value: 'choice1' }).props.onPress();
        });
        expect(mockClose).toBeCalled();
    });

    it('initializes with correct initial state', async () => {
        await act(() => {
            tree.root.findByProps({ value: 'choice1' }).props.onPress();
        });
        expect(
            tree.root.findByProps({ testID: 'valueTxt' }).props.children
        ).toEqual('choice1');
    });

    it('display the correct icon', async () => {
        await act(() => {
            tree.root.findByProps({ value: 'choice1' }).props.onPress();
        });
        expect(
            tree.root.findAllByProps({ testID: 'icon1' }).length
        ).toBeGreaterThan(0);
    });

    it('displays the selected option correctly', async () => {
        await act(() => {
            tree.root.findByProps({ value: 'choice1' }).props.onPress();
        });
        await act(() => {
            tree.root.findByProps({ testID: 'choice2' }).props.onPress();
        });
        expect(mockOnSelect).toBeCalledWith({ id: 2, name: 'choice2' });
    });

    it('handles empty options array', async () => {
        await act(() => {
            tree.update(
                <LogInputDropdown
                    fieldName="DropDown Test"
                    selectedValue={1}
                    labelKey="name"
                    valueKey="id"
                    onSelect={mockOnSelect}
                    options={[]}
                    onOpen={mockOpen}
                    onClose={mockClose}
                />
            );
        });
        await act(() => {
            tree.root
                .findByProps({ testID: 'logInputContainer' })
                .props.onPress();
        });
        expect(tree.root).toBeTruthy();
    });

    it('handles missing onSelect callback', async () => {
        <LogInputDropdown
            fieldName="DropDown Test"
            selectedValue={1}
            labelKey="name"
            valueKey="id"
            options={[
                {
                    id: 1,
                    name: 'choice1'
                },
                {
                    id: 2,
                    name: 'choice2'
                }
            ]}
            onOpen={mockOpen}
            mockClose={mockClose}
        />;
        await act(() => {
            tree.root
                .findByProps({ testID: 'logInputContainer' })
                .props.onPress();
        });
        await act(() => {
            tree.root.findByProps({ testID: 'choice2' }).props.onPress();
        });
        await act(() => {
            tree.root.findByProps({ testID: 'choice1' }).props.onPress();
        });
        expect(mockClose).toBeCalledTimes(3);
    });
});
