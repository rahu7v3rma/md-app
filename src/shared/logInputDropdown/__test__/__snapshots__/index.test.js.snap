// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test log input dropdown snapshot test 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 16,
        "elevation": 5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 8,
        "opacity": 1,
        "paddingHorizontal": 20,
        "paddingVertical": 18,
        "shadowColor": "rgba(0,0,0,0.5)",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 16,
      }
    }
    testID="logInputContainer"
  >
    <Text
      adjustsFontSizeToFit={true}
      style={
        [
          {
            "fontFamily": "Poppins",
          },
          {
            "flex": 1,
            "lineHeight": 20,
          },
          {
            "color": "#58595A",
            "fontSize": 14,
            "fontWeight": "500",
            "lineHeight": 22,
          },
        ]
      }
      testID="fieldNameTxt"
    >
      DropDown Test
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        style={
          [
            {
              "fontFamily": "Poppins",
            },
            {
              "flex": 1,
              "lineHeight": 20,
              "paddingRight": 10,
              "textAlign": "right",
            },
            {
              "color": "#271A51",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 22,
            },
          ]
        }
        testID="valueTxt"
      >
        choice1
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginLeft": 12,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={10}
          bbWidth={15}
          fill="none"
          focusable={false}
          height={10}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "height": "auto",
                "width": "auto",
              },
              {
                "flex": 0,
                "height": 10,
                "width": 15,
              },
            ]
          }
          vbHeight={8}
          vbWidth={12}
          width={15}
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M11 1.5L6 6.5L1 1.5"
              propList={
                [
                  "stroke",
                ]
              }
              stroke={4285700459}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      transparent={true}
      visible={false}
    />
  </View>,
]
`;
